### PR TITLE
Play 2.7.9 & Bootstrap 5.3.0 Upgrades

### PR DESCRIPTION
**Pull request by LetMeTestThat**
# DDCE-2.7.9

- Bootstrap 2.3.0 -> 5.3.0
- Play 2.6.25 -> 2.7.9
- Sbt 1.3.13 -> 1.4.9
- General depedency bumps

 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [x]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/config/wiring.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ class WSHttp @Inject()(httpAuditing: HttpAuditing, wsClient: WSClient, configura
 
 
   def buildRequestWithTimeout(uri: String)(implicit hc: HeaderCarrier): WSRequest = {
```
Comment (DannyPennington97) [inline app/config/wiring.scala:35]: I don't think this method is actually used anywhere, should be safe to remove?

File: app/services/audit/AuditService.scala [hunk 2]
```diff
@@ -43,8 +42,8 @@ class AuditService @Inject()(auditConnector: DefaultAuditConnector,
     )
 
   private def generateTags(session: Session, hc: HeaderCarrier): Map[String, String] =
-    hc.headers.toMap ++
-      hc.headers.toMap ++
+    hc.otherHeaders.toMap ++
```
Comment (DannyPennington97) [inline app/services/audit/AuditService.scala:45]: Does `otherHeaders` contain all the headers needed here?

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -1,27 +1,26 @@
...
+    "uk.gov.hmrc" %% "domain" % "5.11.0-play-27",
+    "uk.gov.hmrc" %% "http-caching-client" % "9.5.0-play-27",
+    "uk.gov.hmrc" %% "tabular-data-validator" % "1.4.0",
     "org.scala-lang.modules" %% "scala-xml" % "1.0.6",
-    "org.scala-lang.modules" %% "scala-parser-combinators" % "1.0.6",
-    "com.typesafe.play" %% "play-json-joda" % "2.6.10",
-    "uk.gov.hmrc" %% "auth-client" % "3.2.0-play-26",
+    "org.scala-lang.modules" %% "scala-parser-combinators" % "1.0.7",
+    "com.typesafe.play" %% "play-json-joda" % "2.6.14",
+    "uk.gov.hmrc" %% "auth-client" % "3.3.0-play-27",
```
Comment (DannyPennington97) [inline project/AppDependencies.scala:18]: auth-client is brought in by bootstrap so it can be removed here I believe

**Comment by DannyPennington97** [comment]
There's a couple of quick win compiler warning fixes to do with the HttpReads implicits and some for the tests too (like the new HttpResponse constructors)

File: app/services/audit/AuditService.scala (hunk 2 above)
Comment (LetMeTestThat) [reply app/services/audit/AuditService.scala:45]: I believe headers was split into `otherHeaders` (those built implicitly) and `extraHeaders` (those added to a request explicitly) so I believe `otherHeaders` is suitable here. Please correct me if I'm wrong though.

File: project/AppDependencies.scala (hunk 3 above)
Comment (LetMeTestThat) [reply project/AppDependencies.scala:18]: Nice, will remove this.

**Review by DannyPennington97 — APPROVED** [review]

**Pull request merged**